### DISCUSSION TITLE
key, value arrays not same length

### DISCUSSION THREAD

**Discussion by Flohlaus** [majo48/connect-web-logger#7] [Q&A]
Hi,

I'd like to scrape the data about my Fröling PE1 on the Fröling connect web.
Unfortunately I get the following error message:
```
2021-10-26 19:14:22 >>> system info
2021-10-26 19:14:24 >>> boiler info
2021-10-26 19:14:25 >>> boiler state name: Standby
2021-10-26 19:14:25 >>> key, value arrays not same length.
```

The key array is len 7 while the value array is only len 2. I count 7 values on the boiler info-page.
![image](https://user-images.githubusercontent.com/79670120/138930212-fca76a95-320e-462f-b757-a814d425aa29.png)

What's behind these lines here? I can't find the typo in the html source. Could this be the reason why I get only 2 values?
https://github.com/majo48/connect-web-logger/blob/94616c53aed0d34cfe715232f61784482210a75b/logger/session.py#L112-L115

**Comment by Flohlaus** [comment]
OK, seems they fixed the typo for 5 of the 7 values.
Maybe this does the trick:
```
            if page_id == 'System':
                values = driver.find_elements_by_xpath("//div[@class='value']")  # proper spelling in html source
            else:
                values1 = driver.find_elements_by_xpath("//div[@class='value']")  # BEWARE: typo in html source
                values2 = driver.find_elements_by_xpath("//div[@calss='value']")  # BEWARE: typo in html source
                values = values1 + values2
```

Now I get one step further, but it's still not running. I'll continue tomorrow...

**Reply by Flohlaus** [reply]
Got it. 
I had to fix some attributes in the database and some titles were not translated in my view. And I added the values for my buffer. 

Thank you very much, what a great tool!! 😍

**Comment by majo48** [comment]
Thank you for bringing this to my attention. This has been fixed in commit https://github.com/majo48/connect-web-logger/commit/65082b27980783c2a6407fa484836a7b28a05d3b.

Sorry for the delay, i was down for two weeks with COVID-19.
